### ISSUE TITLE
Error compiling

### ISSUE THREAD

**Issue by anarkrypto** [nanocurrency/nano-work-server#9]
error[E0502]: cannot borrow `self.cmd.buffer.obj_core` as immutable because it is also borrowed as mutable
    --> /home/anarkrypto/.cargo/registry/src/github.com-1ecc6299db9ec823/ocl-0.18.0/src/standard/buffer.rs:1609:13
     |
1606 |         let (offset, len, queue, flags, ewait, enew, /*is_mapped*/) = self.enq_details()?;
     |                                                                       ---- mutable borrow occurs here
...
1609 |             &self.cmd.buffer.obj_core, true, flags, offset, len, ewait, enew)?;
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^ immutable borrow occurs here
...
1614 |             queue.clone()))
     |             ----- mutable borrow later used here

error[E0502]: cannot borrow `self.cmd.buffer.obj_core` as immutable because it is also borrowed as mutable
    --> /home/anarkrypto/.cargo/registry/src/github.com-1ecc6299db9ec823/ocl-0.18.0/src/standard/buffer.rs:1613:57
     |
1606 |         let (offset, len, queue, flags, ewait, enew, /*is_mapped*/) = self.enq_details()?;
     |                                                                       ---- mutable borrow occurs here
...
1613 |         Ok(MemMap::new(mm_core, len, None, unmap_event, self.cmd.buffer.obj_core.clone(),
     |                                                         ^^^^^^^^^^^^^^^^^^^^^^^^ immutable borrow occurs here
1614 |             queue.clone()))
     |             ----- mutable borrow later used here

error[E0502]: cannot borrow `self.cmd.buffer.obj_core` as immutable because it is also borrowed as mutable
    --> /home/anarkrypto/.cargo/registry/src/github.com-1ecc6299db9ec823/ocl-0.18.0/src/standard/buffer.rs:1635:13
     |
1630 |         let (offset, len, queue, flags, ewait, enew, /*is_mapped*/) = self.enq_details()?;
     |                                                                       ---- mutable borrow occurs here
...
1635 |             &self.cmd.buffer.obj_core, false, flags, offset, len, ewait,
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^ immutable borrow occurs here
...
1646 |             self.cmd.buffer.obj_core.clone(), queue.clone(), /*is_mapped*/))
     |                                               ----- mutable borrow later used here

error[E0502]: cannot borrow `self.cmd.buffer.obj_core` as immutable because it is also borrowed as mutable
    --> /home/anarkrypto/.cargo/registry/src/github.com-1ecc6299db9ec823/ocl-0.18.0/src/standard/buffer.rs:1646:13
     |
1630 |         let (offset, len, queue, flags, ewait, enew, /*is_mapped*/) = self.enq_details()?;
     |                                                                       ---- mutable borrow occurs here
...
1646 |             self.cmd.buffer.obj_core.clone(), queue.clone(), /*is_mapped*/))
     |             ^^^^^^^^^^^^^^^^^^^^^^^^          ----- mutable borrow later used here
     |             |
     |             immutable borrow occurs here

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0502`.

This error indicates that you are trying to borrow a variable as mutable when it
has already been borrowed as immutable.

Erroneous code example:

```
fn bar(x: &mut i32) {}
fn foo(a: &mut i32) {
    let ref y = a; // a is borrowed as immutable.
    bar(a); // error: cannot borrow `*a` as mutable because `a` is also borrowed
            //        as immutable
    println!("{}", y);
}
```

To fix this error, ensure that you don't have any other references to the
variable before trying to access it mutably:

```
fn bar(x: &mut i32) {}
fn foo(a: &mut i32) {
    bar(a);
    let ref y = a; // ok!
    println!("{}", y);
}
```

**rustc 1.40.0 (73528e339 2019-12-16)
Ubuntu 18.04**

**Comment by PlasmaPower** [comment]
Upgrading `ocl` to 0.19.3 (in `Cargo.toml`) fixes this. I'll make a PR when I'm free if I remember.

**Comment by argakiig** [comment]
closed with #11
